crates/oxc_linter/src/rules/react/iframe_missing_sandbox.rs

```diff
@@ -0,0 +1,227 @@
+        .with_label(span)
+}
+
+fn invalid_sandbox_combination_prop(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("An `iframe` element defines a sandbox attribute with both allow-scripts and allow-same-origin which is invalid")
+        .with_label(span)
+}
+
+const ALLOWED_VALUES: Set<&'static str> = phf_set! {
+    "",
+    "allow-downloads-without-user-activation",
+    "allow-downloads",
+    "allow-forms",
+    "allow-modals",
+    "allow-orientation-lock",
+    "allow-pointer-lock",
+    "allow-popups",
+    "allow-popups-to-escape-sandbox",
+    "allow-presentation",
+    "allow-same-origin",
+    "allow-scripts",
+    "allow-storage-access-by-user-activation",
+    "allow-top-navigation",
+    "allow-top-navigation-by-user-activation"
+};
+
+#[derive(Debug, Default, Clone)]
+pub struct IframeMissingSandbox;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Enforce sandbox attribute on iframe elements
+    ///
+    /// ### Why is this bad?
+    ///
+    /// The sandbox attribute enables an extra set of restrictions for the content in the iframe. Using sandbox attribute is considered a good security practice.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```jsx
+    /// <iframe/>;
+    /// <iframe sandbox="invalid-value" />;
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```jsx
+    /// <iframe sandbox="" />;
+    /// <iframe sandbox="allow-origin" />;
+    /// ```
+    IframeMissingSandbox,
+    correctness
+);
+
+impl Rule for IframeMissingSandbox {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.kind() {
+            AstKind::JSXOpeningElement(jsx_el) => {
+                let JSXElementName::Identifier(identifier) = &jsx_el.name else {
+                    return;
+                };
+
+                let name = identifier.name.as_str();
+                if name != "iframe" {
+                    return;
+                }
+
+                has_jsx_prop_ignore_case(jsx_el, "sandbox").map_or_else(
+                    || {
+                        ctx.diagnostic(missing_sandbox_prop(identifier.span));
+                    },
+                    |sandbox_prop| {
+                        validate_sandbox_attribute(sandbox_prop, ctx);
+                    },
+                );
+            }
+            AstKind::CallExpression(call_expr) => {
+                if is_create_element_call(call_expr) {
+                    let Some(Argument::StringLiteral(str)) = call_expr.arguments.first() else {
+                        return;
+                    };
+
+                    if str.value.as_str() != "iframe" {
```

Comment:
`Atom` implements `PartialEq` with `&str`, you shouldn't need to call `as_str()`. If this produces a type error, we'll fix `Atom`'s impls.